std/conv.d

```diff
@@ -4216,6 +4221,20 @@ version(unittest) private class __conv_EmplaceTestClass
     }
 }
 
+unittest // bugzilla 15772
+{
+    abstract class Foo {}
+    class Bar: Foo {}
+    void[] memory;
+    memory.length = typeid(Foo).initializer.length;
```

Comment:
The runtime allocation for `memory` is not necessary, becaue following `emplace` behaviors are tested in compile time.

Also, you can replace the runtime `assert`s with `static assert`s, to make clearer that those are testing the compile-time check of `emplace`.